### DISCUSSION TITLE
NODE_ENV in Dockerfile deps stage

### DISCUSSION THREAD

**Discussion by ahmafi** [vercel/next.js#37442] [Help] [answered]
In the [Dockerfile example](https://github.com/vercel/next.js/blob/4cbc61c740b6f8451cb024f44edc6b5c5ab4aa8a/examples/with-docker-multi-env/docker/production/Dockerfile) for production the ENV is only set in the `runner` stage, shouldn't it be also set in the `deps` stage? So we don't get development tools installed in the production environment? I'm not sure about yarn but setting NODE_ENV works with `npm ci` and it won't install development dependencies.

**Comment by icyJoseph** [comment, marked as answer]
If that's the case, then how would you build the project? For example, if you are using TypeScript, the build step needs `devDependencies` to validate types and compile the code. 

I imagine the same goes for SASS files, and CSS modules which ought to be compiled, to things the browser understands, all of which would require things that we put in the `devDependencies`.

**Reply by ahmafi** [reply]
Yeah, you are completely right, I got confused by my own mistakes. Thanks for the quick response.
